### PR TITLE
fix: wait for the execution and state change sequentially (AR-2377)

### PR DESCRIPTION
**Pull request by trOnk12**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2377" title="AR-2377" target="_blank">BugAR-2377</a>  App crashing when uploading new profile picture through camera
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- The app was crashing because it sometimes could not find the file, because it was being wiped out and overwritten while uploading, because there was another coroutine launched parallel updating the ui state, which would try to access a file that was not exisiting.

### Solutions

- execute sequentailly 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-975/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4411418)(4411418)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4411418)(4411418)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-975/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4412995)(4412995)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4412995)(4412995)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt [hunk 1]
```diff
@@ -35,6 +35,7 @@ import com.wire.kalium.logic.feature.user.SelfServerConfigUseCase
 import com.wire.kalium.logic.feature.user.UpdateSelfAvailabilityStatusUseCase
 import com.wire.kalium.logic.featureFlags.KaliumConfigs
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt]: Is this import being used? 🤔

**Review by alexandreferris — APPROVED** [review]
LGTM! Just have a small comment about an import 🤔

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-975/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4417686)(4417686)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4417686)(4417686)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-975/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4418065)(4418065)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4418065)(4418065)-dev-release.apk)

**Pull request merged**